src/replication/replication_manager.cpp

```diff
@@ -31,10 +31,8 @@ ReplicationManager::~ReplicationManager() = default;
 
 msg_id_t ReplicationManager::GetNextMessageId() {
   msg_id_t next_msg_id = next_msg_id_++;
-  // TODO(WAN): This is not robust to multiple threads attempting to get a new message ID, and may result in message IDs
-  //            getting skipped over. For now, I think this is fine, but find a cheap way of doing this properly?
-  if (next_msg_id_.load() == INVALID_MSG_ID) {
-    next_msg_id_++;
+  if (next_msg_id == INVALID_MSG_ID) {
```

Comment:
As far as I can tell, this will ensure that every message id is unique, no message ids are skipped, and no message id will be equal to `INVALID_MSG_ID`. It could end up assigning "out of order" message ids, in the sense that if thread1 requests a message id before thread2 then thread1 *might* get a higher message id than thread 2. But I don't think that's a problem here.

Also, If `++` is more efficient than `compare_exchange_strong`, it might be worth switching the implementation for `Messenger::GetNextSendCallbackId` to this approach. Though it's probably not too important.